# HG changeset patch
# User Rik <rik@octave.org>
# Date 1400876720 25200
#      Fri May 23 13:25:20 2014 -0700
# Branch stable
# Node ID 9040ad7feb354584c182f9a30b65597b2a2170c2
# Parent  d99475e26c787f2e41a2b458bf79a4d715399ecd
Fix regression in hist with multiple columns and normalization (bug #42394).

* hist.m: Use bsxfun to expand out normalization constant from scalar to vector
if necessary.  Add %!tests for correct behavior.

diff --git a/scripts/plot/draw/hist.m b/scripts/plot/draw/hist.m
--- a/scripts/plot/draw/hist.m
+++ b/scripts/plot/draw/hist.m
@@ -152,17 +152,17 @@ function [nn, xx] = hist (varargin)
              (chist(end,:) - sum (isnan (y)))];
   endif
 
   freq = diff (chist);
 
   if (nargin > 2 && ! ischar (varargin{iarg}))
     ## Normalize the histogram.
     norm = varargin{iarg++};
-    freq *= norm / sum (! isnan (y));
+    freq = bsxfun (@times, freq, norm ./ sum (! isnan (y)));
   endif
 
   if (nargout > 0)
     if (arg_is_vector)  
       ## Matlab compatibility requires a row vector return
       nn = freq';  
       xx = x';
     else
@@ -204,9 +204,11 @@ endfunction
 %!   assert (sum (hist ([1:n], n)), n);
 %!   assert (sum (hist ([1:n], [2:n-1])), n);
 %!   assert (sum (hist ([1:n], [1:n])), n);
 %!   assert (sum (hist ([1:n], 29)), n);
 %!   assert (sum (hist ([1:n], 30)), n);
 %! endfor
 %!assert (hist (1,1), 1)
 %!assert (size (hist (randn (750,240), 200)), [200,240])
-
+## Test bug #42394
+%!assert (isempty (hist (rand (10,2), 0:5, 1)), false)
+%!assert (isempty (hist (rand (10,2), 0:5, [1 1])), false)
